### PR TITLE
add authController and add user model

### PR DESCRIPTION
**Pull request by mojahmed**
Before running the project, make sure to install the necessary dependencies

### CODE REVIEW

**Review by vitaliipp — APPROVED** [review]
Great job! But please don't commit empty files or code that is commented out. You need two approvals on your PR before it can be merged. I suggest that teammates should merge only their own PRs

File: src/app.js
Comment (tianahR) [inline src/app.js:1]: Just a question : why didn't you just rename the file connect.js to connectToDb.js , instead of creating a new file?

File: src/controllers/categoriesController.js
Comment (tianahR) [inline src/controllers/categoriesController.js:1]: Before merging your code, would you please delete this file to avoid conflict, it's an empty file. Thank you

File: src/controllers/commentsController.js
Comment (tianahR) [inline src/controllers/commentsController.js:1]: Same here, before merging, would you please delete this file? Thank you

File: src/controllers/usersController.js
Comment (tianahR) [inline src/controllers/usersController.js:1]: Great job - adding comments !

File: src/routes/categoriesRoute.js
Comment (tianahR) [inline src/routes/categoriesRoute.js:1]: Same here, before merging, would you please delete this file since it's empty and to avoid conflict?

File: src/routes/commentsRoute.js
Comment (tianahR) [inline src/routes/commentsRoute.js:1]: Same here, delete file please

File: src/routes/postsRoute.js
Comment (tianahR) [inline src/routes/postsRoute.js:1]: Samee here

File: src/routes/usersRoute.js
Comment (tianahR) [inline src/routes/usersRoute.js:1]: For naming files in the routes folder, since we have "mainRouter", let's use that same type of naming. For example : usersRouter, postRouter .....

**Review by tianahR — CHANGES_REQUESTED** [review]
Great job ! Before you merge after a mentor's approval, I just requested some minor changes. Thank you

**Comment by butolinka** [comment]
Reviewed, no additions

**Review by butolinka — COMMENTED** [review]
No additions from me

**Review by tianahR — APPROVED** [review]

**Pull request merged**
